mongodb/trigger/mongodb/trigger.go

```diff
@@ -204,11 +204,9 @@ func (evntLsnr *EventListener) listen() {
 			} else {
 				err := evntLsnr.stream.Err()
 				if err != nil {
-					//if err is not nil, it means something bad happened, let's finish our func
-					evntLsnr.logger.Errorf("Error while listening to the MongoDB event stream %s", err)
-					evntLsnr.stream.Close(context.Background())
-					evntLsnr.logger.Infof("Stopped Listening...stream closed")
-					return
+					//if err is not nil, don't stop listening on the event stream in case the connection comes back up.
+					evntLsnr.logger.Errorf("Error while listening to the MongoDB event stream . Detailed Error: %s", err)
```

Comment:
If you are removing the closing of stream here, shouldn't you add it somewhere? Trigger Sop would be ideal.

Reply:
oh yes, you are right.. Will add it to stop() method

Reply:
In addition to this shouldn't you send a "true" to the done channel before closing the stream?